### DISCUSSION TITLE
[High contrast black mode] Installer window background doesn't turn to black

### DISCUSSION THREAD

**Discussion by zoromsftw** [wixtoolset/issues#7279] [Questions]
Using Wix.Corext 3.14.0.1119

Installer main window's background doesn't turn black in 'high contrast black' mode.  Text and Links which turn to white and yellow respectively are not visible on the white background and causing accessibility issue.

**Normal mode:**
![image](https://user-images.githubusercontent.com/128020764/225532308-ced9d71d-7692-4a07-943a-3843aa51c3b6.png)

**High contrast black mode:**
![image](https://user-images.githubusercontent.com/128020764/225532587-734bcd9e-2c5d-4c09-9afc-42a56ce20e52.png)

Accessibility team created this bug on our installer

> When a minimum contrast ratio of 3:1 for UI component is not provided between a component and its background, it is difficult for low-vision users as well as anyone in extreme lighting environments to see and interact with the component.
[MAS 4.3.1 – No Disruption of Accessibility Features .docx (sharepoint.com)](https://microsoft.sharepoint.com/:w:/r/sites/accessibility/_layouts/15/Doc.aspx?sourcedoc=%7b747EC735-D16F-4A28-BE87-C4C6DA9D6E89%7d&file=MAS%204.3.1%20%E2%80%93%20No%20Disruption%20of%20Accessibility%20Features%20.docx&action=default&mobileredirect=true)
![MAS4 3 1_CCA failed for the 'License terms and Privacy notice' links in 'Microsoft Azure AD Connect Provisioning Agent Package' pop-up in black contrast mode](https://user-images.githubusercontent.com/128020764/225534626-9fd9b25b-2bd4-48fe-948b-cab68fc295bc.png)

Is this the expected behaviour? If so, is there any work around?

**Comment by zoromsftw** [comment]
**Additional Information:** Description _(This installation guide will ...)_, Links _(Privacy notice and License terms)_, Checkbox are using same FontId 1 
```
<Font Id="0" Height="-12" Weight="500" Foreground="000000" Background="FFFFFF">Segoe UI</Font>
<Window Width="485" Height="300" HexStyle="100a0000" FontId="0">#(loc.Caption)</Window>
<Font Id="1" Height="-12" Weight="500" Foreground="000000" Background="FFFFFF">Segoe UI</Font>
```
![image](https://user-images.githubusercontent.com/128020764/225597108-43e81d17-ac0d-486c-a939-252106e9ca3d.png)
```
<Font Id="0" Height="-12" Weight="500" Foreground="000000" Background="FFFFFF">Segoe UI</Font>
<Window Width="485" Height="300" HexStyle="100a0000" FontId="0">#(loc.Caption)</Window>
<Font Id="1" Height="-12" Weight="500">Segoe UI</Font>
```
High constrast black mode - Description foreground changed to white but background still remains white (instead of black)
![image](https://user-images.githubusercontent.com/128020764/225597491-43c6343a-0c60-4360-b9cd-774510f098cd.png)
Normal mode - Grey background around checkbox
![image](https://user-images.githubusercontent.com/128020764/225597922-9888217d-d005-4efb-90fa-5ac38ad545d2.png)

Is there any way to turn the whole window's background to black in 'high constrast black' mode?

**Reply by barnson** [reply]
WiX v4 supports system colors like `btnface`. WiX v3 doesn't.

**Reply by barnson** [reply]
A custom BA is the v3 workaround.
